ARCHITECTURE.md

```diff
@@ -0,0 +1,120 @@
+
+`index.js` lists all general actions defined for UI-related settings (such as toggling between themes, text labels, etc); `nodes.js` lists all actions related to the data-fetching and display of node metadata (for the node metadata panel); `pipelines.js` lists all actions related to the data-fetching and updating of the pipeline data, where, on having triggered an async action to fetch pipeline data, all raw pipeline data returned from the API will be further processed and updated in the state via the `loadPipelineData` function in `pipelines.js`. 
+
+`graph.js` lists all actions related to the toggling of elements on the flowchart / pipeline graph. 
+
+### `/reducers`
+We utiilze reducers to format any raw data from the API and update our state according to any triggered actions. Similar to actions, we break down into seperate files for each data type, such as `nodes.js`, `pipeline.js`, etc.
+
+One important concept to note is that the `nodeReducer` in `nodes.js` is where the raw node metadata from the API gets prepared before updating the state. 
+
+### `/selectors`
+For data derived from calculation from the state, we utiilze reducers to perform our calcuations and prepare our data for specfic use cases for different parts of the app. 
+
+For example, `getGraphLayout` within `layout.js` prepares an input object containing all `nodes`, `edges`, `layers`, and status of the graph flag to be imported as a prop into the flowchart component. 
+
+### `/utils`
+The utils folder contains a set of data and functions that are utilized throughout the app. 
+
+`/data` contains a series of json files that serves as mock data for the tests, which along with `state.mock.js`, provides a set of mock data and state that are heavily utilized in the test suites within the app. 
+
+`config.js` provides all fixed configuration values in the app, such as types of flags, sidebar widths, localStroage name, etc. 
+
+`/graph` contains all calculation tools utilized within the flowchart component.
+
+`worker.js` provides all webworker-related utilities (including mocks). 
+
+`flags.js` provides all functions related to the configuration of the feature flag object that acts as a control on experimental features in Kedro-Viz. (For further details please refer to the 'Cross Cutting Concern' section below.)
+
+### `/tools/test-lib` (for react-component import)
+Other than running as a stand-alone webapp, Kedro-viz is also avaliable as an npm package to be imported in any external web project. The `test-lib` directory provides a react project that allows the testing Kedro-viz as a npm package. (the page is accessible via the `npm lib-test` cli command.)
+
+### Jest & React testing library
+All tests within Kedro-viz are set up with Jest, with some tests set up using React testing library to utilize their `fireEvent` function. 
+
+### D3 chart rendering, zoom & chartSize calcs
```

Comment:
@bru5 , let me know if there is anything you think I should elaborate  / expand on for this section. I kept it as simple as possible to avoid the need to constantly main this doc - the new architecture diagram with the actions for the minimap and flowchart should imply a lot of ideas under the hood.

Reply:
resolving this since this section has been rewritten and this comment is not relevant anymore.